projects/batfish/src/main/java/org/batfish/grammar/cisco/CiscoControlPlaneExtractor.java

```diff
@@ -6338,15 +6333,6 @@ public void exitUpdate_source_bgp_tail(Update_source_bgp_tailContext ctx) {
     }
   }
 
-  @Override
-  public void exitUse_af_group_bgp_tail(Use_af_group_bgp_tailContext ctx) {
-    String groupName = ctx.name.getText();
-    if (_currentIpPeerGroup == null && _currentIpv6PeerGroup == null) {
-      throw new BatfishException("Unexpected context for use neighbor group");
-    }
-    _currentPeerGroup.getAfGroups().put(_currentAddressFamily, groupName);
```

Comment:
More info: after removing the last line (write, but never read), the first line is an unused variable. Then after removing the first line, the `if`+`throw` seems useless.